File: 5.c
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#include <stdlib.h>

int main()
{
	int number, i = 0, j = 0;
	char fail;
	
	do 
	{
		printf("Enter a number: ");
		if (scanf("%d", &number))
		 {
			break;
		}
		else if (scanf("%c%*[^\n]", &fail))
		{
			printf("That is not a number. ");
		}
	} while (fail);

	if (number == 0)
	{
		printf("%d\n", number);
		exit(0);
	}
	
	do
	{
		if (i % 3 == 0 || i % 5 == 0)
		{
			printf("%d + ", i);
			j += i;
		}		
		
		i++;

		if (i == number)
		{
			j += i;
			printf("%d = ", i);
		}
	} while (i != number);

	printf("%d\n", j);

	return 0;
}

File: 5.cs
// See https://aka.ms/new-console-template for more information

bool flag;

do
{
    try
    {
        Console.Write("Enter a number: ");
        int num = Convert.ToInt32(Console.ReadLine());
        flag = true;
        if (num == 0)
        {
            Console.WriteLine("");
            Console.WriteLine(num);
            Console.WriteLine("");
            System.Environment.Exit(0);
        }
        checkingReminder(num);
    }
    catch (FormatException f_e)
    {
        Console.WriteLine("Not a number! " + f_e.Message + "\n");
        flag = false;
    }

} while (flag == false);

void checkingReminder(int NUM)
{
    int i = 0, j = 0;

    do
    {
        if (i % 3 == 0 || i % 5 == 0)
        {
            Console.Write(i + " + ");
            j += i;
        }
        i++;
    } while (i != NUM);

    j += i;

    Console.WriteLine(i + " = " + j);
}
